aptos-move/aptos-vm/src/sharded_block_executor/local_executor_shard.rs

```diff
@@ -0,0 +1,249 @@
+        num_shards: usize,
+        num_threads: usize,
+        command_rx: Receiver<ExecutorShardCommand<S>>,
+        result_tx: Sender<Result<Vec<Vec<TransactionOutput>>, VMStatus>>,
+        cross_shard_txs: Vec<Vec<Sender<CrossShardMsg>>>,
+        cross_shard_rxs: Vec<Receiver<CrossShardMsg>>,
+    ) -> Self {
+        let cross_shard_client =
+            Arc::new(LocalCrossShardClient::new(cross_shard_txs, cross_shard_rxs));
+        let coordinator_client = Arc::new(LocalCoordinatorClient::new(command_rx, result_tx));
+        let executor_service = Arc::new(ShardedExecutorService::new(
+            shard_id,
+            num_shards,
+            num_threads,
+            coordinator_client,
+            cross_shard_client,
+        ));
+        let executor_service_clone = Arc::clone(&executor_service);
+        let join_handle = thread::Builder::new()
+            .name(format!("executor-shard-{}", shard_id))
+            .spawn(move || executor_service_clone.start())
+            .unwrap();
+        Self {
+            _executor_service: executor_service,
+            join_handle: Some(join_handle),
+        }
+    }
+
+    pub fn setup_local_executor_shards(
+        num_shards: usize,
+        num_threads: Option<usize>,
+    ) -> LocalExecutorClient<S> {
+        let num_threads = num_threads
+            .unwrap_or_else(|| (num_cpus::get() as f64 / num_shards as f64).ceil() as usize);
+        let mut cross_shard_msg_txs = vec![];
+        let mut cross_shard_msg_rxs = vec![];
+        let mut command_txs = vec![];
+        let mut command_rxs = vec![];
+        let mut result_txs = vec![];
+        let mut result_rxs = vec![];
+        for _ in 0..num_shards {
+            let (command_tx, command_rx) = unbounded();
+            let (result_tx, result_rx) = unbounded();
+            command_rxs.push(command_rx);
+            result_txs.push(result_tx);
+            command_txs.push(command_tx);
+            result_rxs.push(result_rx);
+            let mut current_shard_msg_txs = vec![];
+            let mut current_shard_msg_rxs = vec![];
+            for _ in 0..MAX_ALLOWED_PARTITIONING_ROUNDS {
+                let (messages_tx, messages_rx) = unbounded();
+                current_shard_msg_txs.push(messages_tx);
+                current_shard_msg_rxs.push(messages_rx);
+            }
+            cross_shard_msg_txs.push(current_shard_msg_txs);
+            cross_shard_msg_rxs.push(current_shard_msg_rxs);
+        }
+        let executor_shards = command_rxs
+            .into_iter()
+            .zip(result_txs.into_iter())
+            .zip(cross_shard_msg_rxs.into_iter())
+            .enumerate()
+            .map(|(shard_id, ((command_rx, result_tx), cross_shard_rxs))| {
+                Self::new(
+                    shard_id as ShardId,
+                    num_shards,
+                    num_threads,
+                    command_rx,
+                    result_tx,
+                    cross_shard_msg_txs.clone(),
```

Comment:
let's create the LocalCrossShardClient and pass it instead of the last two arguments?